### DISCUSSION TITLE
Protection against authentication bypass exploits

### DISCUSSION THREAD

**Discussion by DasPhero** [gravitational/teleport#10993] [Q&A]
Heyah,

we were discussing to use teleport for our network, but some concerns were raised about the proxy being a "single point of failure (security wise)" as all daemons connect back to it.
Are there any mechanims in place to prevent an attacker from accessing arbitrary servers with elevated privileges for example by bypassing the authentication server using a potential zero day exploit after an update?

**Comment by webvictim** [comment]
There are various safeguards and defenses in depth built into Teleport to try and prevent authentication bypasses. You can see details of our routine security audits published on our website - https://goteleport.com/resources/audits/

You'll hopefully notice that any kinds of bypasses which have been detected and fixed during prior audits would require a very high level of access i.e. you must be physically logged onto the Teleport auth server itself to execute them. We haven't yet had knowledge of a bypass which is able to be exploited remotely.
